designs/credentialManagement/credentialManagement.md

```diff
@@ -0,0 +1,139 @@
+
+1. `CredentialIdentifier` - Represents the globally unique identifier for a possible credential profile in the toolkit. This identifier must be deterministic 
+and represent the same underlying source across IDE sessions. This is shown to the user as the **Profile** in the UI.
+
+1. `AwsCredentialsProvider` - SDK interface that resolves AWS Credentials from the provider. For more info, see [AwsCredentialsProvider] in the SDK.
+
+1. `ToolkitCredentialsProvider` - A class that implements `AwsCredentialsProvider`. This class does not
+do any actual resolving of credentials, but instead leaves it to another concrete implementation by delegating to another implementation of 
+`AwsCredentialsProvider`. This class works as a bridge between a `CredentialIdentifier` and a `AwsCredentialsProvider`.
+
+1. `CredentialProviderFactory` - Abstract class that knows and understands one "category" of credentials (e.g. Shared Credentials files) and has two jobs.
+ 
+    1. Detect valid `CredentialIdentifier`s that should be presented to the user as a possible choice to use in the IDE. A `CredentialIdentifier` is determined 
+    to be valid if and only if the credential source has all the required information to comply with the underlying credential source's contract. For example, 
+    a factory that handles static credentials would need to make sure that both access and secret keys are provided. 
+
+        **It MUST not verify if the credentials themselves are valid (able to make an AWS call) at creation time of the `CredentialIdentifier`.**
+
+    1. Convert the `CredentialIdentifier` it created earlier iinto a valid `AwsCredentialProvider`. This is implementation specific and may re-use existing 
+    `AwsCredentialProvider` implementations.
+
+1. `CredentialManager` - Acts as the entry point to the credential system for the rest of the toolkit. Its 
+job is to keep track of all `CredentialIdentifier` that should be presented to the user as a possible profile they can use. 
+
+### Extension System
+
+In order to make the credential management subsystem extendable, we leverage the IntelliJ [Extension Point] system. Whenever `CredentialManager` needs to 
+communicate with a `CredentialProviderFactory`, it queries the Extension Point `aws.toolkit.credentialProviderFactory` for implementations and proceeds to 
+search the list for the factory with the requested ID.
+
+Example of usage:
+```xml
+<extensions defaultExtensionNs="aws.toolkit">
+    <credentialProviderFactory implementation="software.aws.toolkits.jetbrains.core.credentials.profiles.ProfileCredentialProviderFactory"/>
+</extensions>
+```
+
+### Deep Credential Validation
+
+In order to validate the credentials returned by the `ToolkitCredentialsProvider` are able to mak AWS calls, we make a call to `sts::getCallerIdentity`. 
+If the call fails, we consider the credentials to be invalid.
+
+## Connection Settings Management
+
+The class [AwsConnectionManager] is the entry point into this system.
```

Comment:
Was this renamed? or is this new in the connection-ux branch?

Reply:
Yes, it was renamed in the connection-ux branch. These names are the current names according to that branch.

It was ProjectAccountSettingsManager